pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -156,9 +158,9 @@ def create_csr_filter(self, cfg):
         elif cfg.enable_access_invalid_csr_level:
             self.include_reg = cfg.invalid_priv_mode_csrs
         else:
-            if cfg.init_privileged_mode == "MACHINE_MODE":      # Machine Mode
+            if cfg.init_privileged_mode == privileged_mode_t.MACHINE_MODE:      # Machine Mode
                 self.include_reg.append("MSCRATCH")
-            elif cfg.init_privileged_mode == "SUPERVISOR_MODE":  # Supervisor Mode
+            elif cfg.init_privileged_mode == privileged_mode_t.SUPERVISOR_MODE:  # Supervisor Mode
```

Comment:
Hi @aneels3 ,
I believe above three lines should use `priviledged_mode_t.X_MODE` instead of string `"X_MODE"`. Would you please confirm it?

Reply:
Hi @hodjat91 
Thanks for this. I forgot to do changes for this in the previous #701 PR.
LGTM.